File: backend/app/schemas/trading.py
"""交易 Schemas"""

from datetime import datetime
from decimal import Decimal
from pydantic import BaseModel, Field
from uuid import UUID


class TradingAccountCreate(BaseModel):
    """创建交易账户"""

    account_type: str = Field(default="paper", pattern="^(paper|live)$")
    market: str = Field(default="CN", pattern="^(CN|HK|US)$")
    initial_cash: Decimal = Field(default=Decimal("1000000"), gt=0)


class TradingAccountResponse(BaseModel):
    """交易账户响应"""

    id: UUID
    user_id: UUID
    account_type: str
    cash_cny: Decimal
    cash_hkd: Decimal
    cash_usd: Decimal
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PositionResponse(BaseModel):
    """持仓响应"""

    id: UUID
    account_id: UUID
    symbol: str
    market: str
    quantity: int
    avg_cost: Decimal
    current_price: Decimal | None
    unrealized_pnl: Decimal | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class OrderCreate(BaseModel):
    """创建订单"""

    symbol: str = Field(..., min_length=1, max_length=20)
    market: str = Field(..., pattern="^(CN|HK|US)$")
    side: str = Field(..., pattern="^(buy|sell)$")
    order_type: str = Field(..., pattern="^(market|limit)$")
    quantity: int = Field(..., gt=0)
    price: Decimal | None = Field(None, gt=0)


class OrderResponse(BaseModel):
    """订单响应"""

    id: UUID
    account_id: UUID
    symbol: str
    market: str
    side: str
    order_type: str
    quantity: int
    price: Decimal | None
    filled_quantity: int
    filled_price: Decimal | None
    status: str
    filled_at: datetime | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/trading.ts
/**
 * 交易相关类型定义
 */

import type { MarketType } from './common'

export type OrderSide = 'buy' | 'sell'
export type OrderType = 'market' | 'limit'
export type OrderStatus = 'pending' | 'filled' | 'cancelled' | 'rejected'

export interface TradingAccount {
  id: string
  user_id: string
  account_type: 'paper'
  cash_cny: number
  cash_hkd: number
  cash_usd: number
  created_at: string
  updated_at: string
}

export interface Position {
  id: string
  account_id: string
  symbol: string
  market: MarketType
  quantity: number
  avg_cost: number
  current_price: number
  unrealized_pnl: number
  created_at: string
  updated_at: string
}

export interface Order {
  id: string
  account_id: string
  symbol: string
  market: MarketType
  side: OrderSide
  order_type: OrderType
  quantity: number
  price?: number
  filled_quantity: number
  filled_price?: number
  status: OrderStatus
  filled_at?: string
  created_at: string
}

export interface CreateOrderRequest {
  symbol: string
  market: MarketType
  side: OrderSide
  order_type: OrderType
  quantity: number
  price?: number
}

export interface PortfolioOverview {
  total_value: number
  total_pnl: number
  return_rate: number
  positions_count: number
}
